arbnode/batch_poster.go

```diff
@@ -585,23 +585,29 @@ func (b *BatchPoster) ParentChainIsUsingEIP7623(ctx context.Context, latestHeade
 		data = append(data, 1)
 	}
 
-	gas1, err := estimateGas(rpcClient.Client(), ctx, estimateGasParams{
+	var blockHex string
+	err := rpcClient.Client().CallContext(ctx, &blockHex, "eth_blockNumber")
```

Comment:
Here there's no reason to use the raw RPC. The original "rpcClient" has direct functions you can use. Also - going with the latest block increases chance of reorgs and weird results.
My first instinct is to look at the "Safe block", but that could be quite far back for L3s.. maybe just go for latest - some small constant offset (5?). It's not that terrible if this block does indeed get reorged.

A final note: we should have a config for this behavior.. something like parent-chain-eip7623 "on"/"off"/"auto" - where only auto enables checking dynamically, and even auto stops checking and returns true after the first time it detects that it was activated. (this can be a separate PR but should be in the next rc)

Reply:
done